cognite/extractorutils/configtools.py

```diff
@@ -503,9 +503,20 @@ def load_yaml(
         InvalidConfigError: If any config field is given as an invalid type, is missing or is unknown
     """
 
-    def env_constructor(_, node):
-        # Expnadvars uses same syntax as our env var substitution
-        return os.path.expandvars(node.value)
+    def env_constructor(_: yaml.SafeLoader, node):
+        bool_values = {
+            "yes": True,
+            "no": False,
+            "true": True,
+            "false": False,
+            "on": True,
+            "off": False,
+        }
+        expanded_value = os.path.expandvars(node.value)
+        return bool_values.get(expanded_value.lower(), node.value)
```

Comment:
Honestly, this is not the ideal solution. We want to replace env vars first and then call default `yaml `constructors for type inference.  But haven't seen a way to enforce our `env_contructor` to run first so this is just a solution for `env var str -> bool`. Numeric conversions, and the others still will not work. 

Maybe we should manipulate `source` and find/replace env var syntaxes but idk.